script.js

```diff
@@ -317,12 +325,72 @@ const treeTable = {
     }
 }
 
+const categoriesTable = {
+    rows: [
+        {
+            view: "toolbar",
+            elements: [
+                {
+                    view: "text",
+                    id: "category_input",
+                    placeholder: "Category name",
+                },
+                {
+                    view: "button",
+                    label: "Add Category",
+                    id: "btn_add_category",
+                    autowidth: true,
+                    css: "webix_primary",
+                    click: function () {
+                        const genre = $$("category_input").getValue();
+                        if (genre === "") {
+                            webix.message(`Categoty should not be empty`);
+                        } else {
+                            categories.add({ value: genre }, 0);
+                            webix.message(`New category added`);
+                            $$("category_input").setValue("");
+                        }
+                    }
+                },
+                {
+                    view: "button",
+                    label: "Delete Category",
+                    id: "btn_delete_category",
+                    autowidth: true,
+                    css: "webix_primary",
+                    click: function () {
+                        const id = $$("categoriesTable").getSelectedId();
+                        if (id) {
+                            categories.remove(id);
+                            webix.message(`Category deleted`);
+                        }
+                    }
+                },
+            ]
+        },
+        {
+            view: "datatable",
+            id: "categoriesTable",
+            autoConfig: true,
```

Comment:
don't use both autoConfig and columns, as they contradict each other

Reply:
Will keep it in mind too.